tests/Integration/Behaviour/Features/Context/ShopFeatureContext.php

```diff
@@ -65,16 +71,94 @@ public function shopWithNameExists(string $reference, string $shopName)
     /**
      * @Given single shop context is loaded
      */
-    public function singleShopContextIsLoaded()
+    public function singleShopContextIsLoaded(): void
     {
         Shop::setContext(Shop::CONTEXT_SHOP, Configuration::get('PS_SHOP_DEFAULT'));
     }
 
     /**
      * @Given multiple shop context is loaded
      */
-    public function multipleShopContextIsLoaded()
+    public function multipleShopContextIsLoaded(): void
     {
         Shop::setContext(Shop::CONTEXT_ALL);
     }
+
+    /**
+     * @Given I add a shop :reference with name :shopName for the group :shopGroupName
+     *
+     * @param string $reference
+     * @param string $shopName
+     * @param string $shopGroupName
+     */
+    public function addShop(string $reference, string $shopName, string $shopGroupName): void
+    {
+        $shop = new Shop();
+        $shop->active = true;
+        $shop->id_shop_group = ShopGroup::getIdByName($shopGroupName);
+        $shop->id_category = 2;
+        $shop->theme_name = _THEME_NAME_;
+        $shop->name = $shopName;
+        if (!$shop->add()) {
+            throw new RuntimeException(sprintf('Could not create shop: %s', Db::getInstance()->getMsgError()));
+        }
+        $shop->setTheme();
+
+        SharedStorage::getStorage()->set($reference, $shop);
+    }
+
+    /**
+     * @When I search for shops with the term :searchTerm I should get the following results:
+     *
+     * @param string $searchTerm
+     * @param TableNode $table
+     */
+    public function assertFoundShops(string $searchTerm, TableNode $table): void
+    {
+        $expectedShops = $table->getColumnsHash();
+        $foundShops = $this->getQueryBus()->handle(new SearchShops($searchTerm));
+
+        foreach ($expectedShops as $key => $currentExpectedShop) {
+            $wasCurrentExpectedShopFound = false;
+            foreach ($foundShops as $currentFoundShop) {
+                if (strtolower($currentExpectedShop['name']) == strtolower($currentFoundShop->getName())) {
```

Comment:
Why do you need to compare with lower keys? You could check strict equality to make sure the real shop name is returned no?

Reply:
Nice catch, it's a mistake. (I compare lower keys when I compare user input, but that's not the case here)